wolfcrypt/src/port/Renesas/renesas_tsip_aes.c

```diff
@@ -40,11 +40,101 @@
 #include "wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h"
 
 struct Aes;
+/*
+ * This key store keeps the key-index which TSIP generated.
+ * The key will not be used for TLS communication but crypt-test and
+ * benchmark purpose only.
+ */
+static TSIP_AES_CTX  tsip_AES_KeyStore;
+
+
+/*
+ * tsip_AesGenKey
+ *
+ This API is intended to generate an AES key and share it between the Aes
+ structures for encryption and the one for decryption. Key share is done
+ via the tsip_AES_keyStore variable and dir (the fifth parameter of this
+  function).
+ The expected use case is as follows:
+ 1. wc_AesSetKey API is called to set raw AES key and iv to the given
+    Aes structure.
+ 2. tsip_AesGenKey is called underneath the API. When "dir" is
+    AES_ENCRYPTION, it generate a AES key of the specified key length then
+    stores it into aes->ctx with other info. aes->ctx is also stored into
+    tsip_AES_keyStore variable. The given raw AES key will not be used nor
+    stored.
+ 3. Data encryption will be performed with the key generated in 2.
+ 4. wc_AesSetKey API is called to set the raw AES key and iv to the given
+    Aes structure for decryption with "dir" is AES_DECRYPTION. In this case,
+    Aes key is not generated but restored from tsip_AES_keyStore variable
+    with other information.
+ 5. Data decryption will be performed with the shared AES key.
+
+ Above use case occurs only in crypt-test and benchmark.
+ */
+int tsip_AesGenKey(struct Aes* aes, word32 keylen, const byte* iv, int dir)
```

Comment:
Can we rename this function? Generate key indicates creation of a new key. This appears to be loading the key. Perhaps `tsip_AesLoadKey`?

Reply:
As you pointed out, this function does not always generate a key. But, when the 4th argument is AES_ENCRYPTION, the R_TSIP_GenerateAesXXXRandomKeyIndex () function is called internally to actually generate and store the key.